### PR TITLE
Fix weird error message when importing certificate with no password

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
Also do not prompt for password when importing .cer file
Fixes http://teampulse.telerik.com/view#item/292671
Merge along with https://github.com/telerik/mobile-cli-lib/pull/348

### CODE REVIEW

**Comment by Mitko-Kerezov** [comment]
run ci

File: lib/commands/cryptographic-identities.ts [hunk 1]
```diff
@@ -605,12 +605,12 @@ export class ImportCryptographicIdentity implements ICommand {
 					this.$logger.error(err.message + os.EOL + "Verify that you have provided the correct file and password and try again.");
 				}
 
-				if (noErrorOccurred) {
+				if (noErrorOccurred || importType === CryptographicIdentityConstants.X509CERTIFICATE) {
```
Comment (rosen-vladimirov) [inline lib/commands/cryptographic-identities.ts]: I'm confused... If something's wrong, noErrorOccurred will be false, but we'll still break the for and a messages "Successfully imported.... " will be shown to the user. What am I missing here?

File: lib/commands/cryptographic-identities.ts [hunk 2]
```diff
@@ -605,12 +605,12 @@ export class ImportCryptographicIdentity implements ICommand {
...
 					break;
 				}
 			}
 
-			if (!noErrorOccurred) {
+			if (!noErrorOccurred && importType === CryptographicIdentityConstants.PKCS12CERTIFICATE) {
```
Comment (rosen-vladimirov) [inline lib/commands/cryptographic-identities.ts]: more confusing - if there's an error, but the certificate is not PKCS12, we'll not fail, again we'll show that the certificate is imported successfully.

File: lib/commands/cryptographic-identities.ts [hunk 3]
```diff
@@ -614,10 +624,19 @@ export class ImportCryptographicIdentity implements ICommand {
...
+		}).future<Server.CryptographicIdentityData[]>()();
+	}
+
+	private ImportCertificateWithoutPassword(importType: string, certificateFile: string): IFuture<Server.CryptographicIdentityData[]> {
+		return (() => {
+			try {
+				let targetFile = this.$fs.createReadStream(certificateFile);
+				return this.$server.identityStore.importIdentity(<any>importType, '', targetFile).wait();
+			} catch (error) {
+				this.$errors.failWithoutHelp(error.message);
```
Comment (teobugslayer) [inline lib/commands/cryptographic-identities.ts]: Is this needed?

Comment (Mitko-Kerezov) [inline lib/commands/cryptographic-identities.ts]: Else you get a message which more often than not says there is no request for this certificate after which you get the help for the command.
Given the fact that you've used the command properly I don't think it's suitable to display help

**Comment by Mitko-Kerezov** [comment]
run ci

**Comment by Mitko-Kerezov** [comment]
run ci

File: lib/commands/cryptographic-identities.ts [hunk 4]
```diff
@@ -566,29 +566,39 @@ export class ImportCryptographicIdentity implements ICommand {
...
+			_.each(result, identity => {
+				this.$logger.info("Imported certificate '%s'.", identity.Alias);
+			});
+		}).future<void>()();
+	}
 
-			let targetFile : any;
-			let result : Server.CryptographicIdentityData[];
-			let noErrorOccurred : boolean;
+	private ImportCertificateWithPassword(importType: string, password: string, certificateFile: string): IFuture<Server.CryptographicIdentityData[]> {
```
Comment (rosen-vladimirov) [inline lib/commands/cryptographic-identities.ts]: method name is starting with Capital letter 
:crying_cat_face:

File: lib/commands/cryptographic-identities.ts [hunk 5]
```diff
@@ -614,10 +624,19 @@ export class ImportCryptographicIdentity implements ICommand {
...
 
-			_.each(result, identity => {
-				this.$logger.info("Imported certificate '%s'.", identity.Alias);
-			});
-		}).future<void>()();
+			return result;
+		}).future<Server.CryptographicIdentityData[]>()();
+	}
+
+	private ImportCertificateWithoutPassword(importType: string, certificateFile: string): IFuture<Server.CryptographicIdentityData[]> {
```
Comment (rosen-vladimirov) [inline lib/commands/cryptographic-identities.ts]: one more method name starting with capital letter 
:crying_cat_face: :crying_cat_face:

File: lib/commands/cryptographic-identities.ts [hunk 6]
```diff
@@ -566,29 +566,39 @@ export class ImportCryptographicIdentity implements ICommand {
...
 			let importType = extension === ".p12" ? CryptographicIdentityConstants.PKCS12CERTIFICATE : CryptographicIdentityConstants.X509CERTIFICATE;
 
 			if(!this.$fs.exists(certificateFile).wait()) {
 				this.$errors.fail("The file '%s' does not exist.", certificateFile);
 			}
 
-			if(!password && importType === CryptographicIdentityConstants.PKCS12CERTIFICATE) {
-				isPasswordRequired = true;
-			}
+
```
Comment (rosen-vladimirov) [inline lib/commands/cryptographic-identities.ts]: additional blank line, which is not required

**Comment by rosen-vladimirov** [comment]
:+1:

**Pull request merged**